flux2/templates/kustomize-controller.yaml

```diff
@@ -19,6 +19,9 @@ spec:
       annotations:
         prometheus.io/port: "8080"
         prometheus.io/scrape: "true"
+        {{- if .Values.podAnnotations }}
```

Comment:
same here, you might have a problem if .Values.podAnnotations will include the prometheus annotations, maybe consider to move the prometheus to the code (getFluxValues) and do the override merge there?